clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -2397,7 +2399,12 @@ public ProducerRecord<byte[], byte[]> onSend(ProducerRecord<byte[], byte[]> reco
         }
 
         @Override
-        public void onAcknowledgement(RecordMetadata metadata, Exception exception) {
+        public void onAcknowledgement(RecordMetadata metadata, Exception exception, Headers headers) {
+            if (!(headers instanceof RecordHeaders)) {
```

Comment:
For what cases do we expect this to be "false"? -- given that it is test code, I am wondering if we need it, or if it could (in the worst case) even mask a potential bug?

Reply:
Yes, I agree.

Reply:
Good call, I will remove it

Reply:
removed at 988c09f481